### PR TITLE
Update Cross Entropy Loss for Reduced Memory Usage

### PR DESCRIPTION
**Pull request by ozan-oktay**

### CODE REVIEW

**Review by ant0nsc — APPROVED** [review]

File: InnerEye/ML/models/losses/cross_entropy.py [hunk 1]
```diff
@@ -130,32 +128,22 @@ def forward_minibatch(self, output: torch.Tensor, target: torch.Tensor, **kwargs
...
-        # Compute log posterior probabilities
-        log_prob = F.log_softmax(output, 1)
-
-        axes = list(range(2, len(output.shape)))
+        # Determine class weights for unbalanced datasets
+        if self.class_weight_power is not None and self.class_weight_power != 0.0:
+            class_weight = get_class_weights(target, class_weight_power=self.class_weight_power) 
+
+        # Compute negative log-likelihood
+        log_prop = F.log_softmax(output, dim=1)
```
Comment (Shruthi42) [inline InnerEye/ML/models/losses/cross_entropy.py:136]: log_prob?

**Review by Shruthi42 — APPROVED** [review]

File: InnerEye/ML/models/losses/cross_entropy.py (hunk 1 above)
Comment (ozan-oktay) [reply InnerEye/ML/models/losses/cross_entropy.py:136]: good one 👍

**Review by Shruthi42 — APPROVED** [review]

File: Tests/ML/models/losses/test_cross_entropy_loss.py [hunk 2]
```diff
@@ -104,7 +104,7 @@ def test_cross_entropy_loss_integration(focal_loss_gamma: float, loss_upper_boun
     # Define a basic model (We actually don't even a non-linear unit to solve it)
     net = ToyNet()
     opt = optim.Adam(net.parameters(), lr=0.001, betas=(0.9, 0.999))
-    loss_fn = CrossEntropyLoss(class_weight_power=1.0, focal_loss_gamma=focal_loss_gamma)
+    loss_fn = CrossEntropyLoss(class_weight_power=1.5, focal_loss_gamma=focal_loss_gamma)
```
Comment (javier-alvarez) [inline Tests/ML/models/losses/test_cross_entropy_loss.py:107]: why do we need to change this?

Comment (ozan-oktay) [reply Tests/ML/models/losses/test_cross_entropy_loss.py]: Wanted to try out a value larger than 1.0 to test the class_weighting function since it's not covered in the test. With the new commit, I just added as another test instance. (updated)

**Review by Shruthi42 — APPROVED** [review]

**Review by ant0nsc — APPROVED** [review]
